### PR TITLE
unix, windows: set non-block mode in uv_poll_init

### PR DESCRIPTION
**Pull request by saghul**
R=@bnoordhuis
R=@piscisaureus

### CODE REVIEW

File: test/test-poll.c [hunk 1]
```diff
@@ -86,22 +85,7 @@ static int got_eagain(void) {
...
-  ASSERT(flags >= 0);
-  r = fcntl(sock, F_SETFL, flags | O_NONBLOCK);
-  ASSERT(r >= 0);
-#endif
-}
-
-
-static uv_os_sock_t create_nonblocking_bound_socket(
+static uv_os_sock_t create_bound_socket(
     struct sockaddr_in bind_addr) {
```
Comment (bnoordhuis) [inline test/test-poll.c]: Minor style nit: this now fits on a single line.

**Comment by bnoordhuis** [comment]
CI! https://jenkins-iojs.nodesource.com/view/libuv/job/libuv+any-pr+multi/50/

**Comment by bnoordhuis** [comment]
LGTM if the CI is green (or at least, not redder than it was before) but maybe the commit log can document the motivation for the change.

**Comment by saghul** [comment]
I added some ratinale to the commit log and fixed the long line. The reds in the build don't look related to this.

**Comment by bnoordhuis** [comment]
LGTM

**Comment by saghul** [comment]
Landed in b30a3e6, thanks!

**Pull request closed without merging**